### ISSUE TITLE
tf.transpose crashes(abort) if `a` is complex

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#46891]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
`tf.transpose` crashes(abort) if `a` is complex and `conjugate`=True

**Describe the expected behavior**
expect no crash

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
tf.transpose(conjugate=True, a=complex(1))
~~~

Output:
~~~python
2021-02-03 17:58:05.565680: F ./tensorflow/core/kernels/transpose_functor.h:169] Check failed: in.dims() >= 2 (0 vs. 2)
Aborted (core dumped)
~~~

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3, TF v2.4 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/3beaf907cf0caf938eefcab8d92def67/46891.ipynb). Thanks!

**Comment by yongtang** [comment]
Added PR #46973 for the fix.

**Comment by oawxkw** [comment]
I also observed the following API aliases can cause the same issue in older versions of tensorflow.
Users should be cautious when using them on both CPU and GPU up to tensorflow 2.4.0 (v2.4.0-rc4-71-g582c8d236cb).

- `(tf.transpose)`, `tf.compat.v1.transpose`

<details>
  <summary>Code to reproduce the issue in <code>tf.compat.v1.transpose</code> in older versions</summary>

```python
import tensorflow as tf
print(tf.version.GIT_VERSION, tf.version.VERSION, flush=True)
print(tf.config.list_physical_devices(), flush=True)


try:
    tf.compat.v1.transpose(conjugate=True, a=complex(1))
except Exception as e:
    print("Error:", str(e), flush=True)
print("Success!", flush=True)
```

On GPU, the Check failed error occurs:

```text
v2.4.0-rc4-71-g582c8d236cb 2.4.0
[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'), PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]
2023-09-08 10:49:02.743006: F ./tensorflow/core/kernels/transpose_functor.h:169] Check failed: in.dims() >= 2 (0 vs. 2)
Aborted (core dumped)
```

This behavior is also reproducible on my CPU machine:

```text
v2.4.0-rc4-71-g582c8d236cb 2.4.0
[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU')]
2023-09-08 10:48:56.754273: F ./tensorflow/core/kernels/transpose_functor.h:169] Check failed: in.dims() >= 2 (0 vs. 2)
Aborted (core dumped)
```
</details>

It seems to be fixed in tensorflow 2.4.3 (v2.4.2-142-g72bb4c22adb) and later versions.
